opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -43,22 +58,28 @@ impl<T: Number<T>> ValueMap<T> {
             self.no_attribute_value.add(measurement);
             self.has_no_value_attribute_value
                 .store(true, Ordering::Release);
-        } else if let Ok(mut values) = self.values.lock() {
-            let size = values.len();
-            match values.entry(attrs) {
-                Entry::Occupied(mut occupied_entry) => {
-                    let sum = occupied_entry.get_mut();
-                    *sum += measurement;
-                }
-                Entry::Vacant(vacant_entry) => {
-                    if is_under_cardinality_limit(size) {
-                        vacant_entry.insert(measurement);
-                    } else {
-                        values
-                            .entry(STREAM_OVERFLOW_ATTRIBUTE_SET.clone())
-                            .and_modify(|val| *val += measurement)
-                            .or_insert(measurement);
-                        global::handle_error(MetricsError::Other("Warning: Maximum data points for metric stream exceeded. Entry added to overflow.".into()));
+        } else {
+            let bucket_key = Self::hash_to_bucket(&attrs);
+            if let Some(bucket) = self.buckets.get(&bucket_key) {
+                let mut values = bucket.lock().unwrap();
+                let size = values.len();
+                match values.entry(attrs) {
+                    Entry::Occupied(mut occupied_entry) => {
+                        let sum = occupied_entry.get_mut();
+                        *sum += measurement;
+                    }
+                    Entry::Vacant(vacant_entry) => {
+                        if is_under_cardinality_limit(size) {
+                            vacant_entry.insert(measurement);
+                            self.total_count.fetch_add(1, Ordering::SeqCst);
```

Comment:
Do we need `SeqCst` here? Given we only +1 here the order of which happens first doesn't really matter that much?

Reply:
Yes, good point. We can make it to `Ordering::Relaxed`. I'm also exploring possibilities to eliminate this `total_count` variable. Despite being atomic, it still encounters thread contention. The goal of this count is to reserve the size of the output vector in the collection phase.

Reply:
`total_count` (used to maintain the data-points count) has been removed, and the count is calculated by iteration over the buckets during the collect cycle.